an-rope/src/internals.rs

```diff
@@ -55,20 +55,38 @@ impl BranchNode {
     fn split(self, index: usize) -> (Node, Node) {
         let weight = (&self).weight;
         if index < weight {
-            // split the left node
+            // if the index to split on is less than this node's weight,
+            // we split this node's left child.
             let (left, left_right) = self.left.split(index);
+            // the left side of the split left child will become the left side
+            // of the split pair.
+
             let right = if (&left_right).len() == 0 {
+                // if the right side of the split is empty, then the right
+                // side of the returned pair is just this node's right child
                 *self.right
             } else {
+                // otherwise, the right side of the returned pair is a branch
+                // containing the right side of the split node on the left,
+                // and this node's right child on the right
                 Node::new_branch(left_right, *self.right)
             };
             (left, right)
```

Comment:
The return type is a tuple because, ifyou look at how `insert` works, or at `Rope::split`, the two primary consumers of this method, the new Nodes returned from `split` don't end up branched together.

In `insert`, we concat the rope to insert onto the left side of the split, and then concat the right side onto the result of that concatenation. so the left and right side of the split don't actually go in a branch together. If we returned them in a `Branch`, the caller would just move the nodes back out of that branch. We'd spend extra time boxing and then unboxing the nodes. 

Also I don't think it's semantic to put the split nodes in a branch, because the whole point of `split`ing is to separate them.